library/std/src/net/ip.rs

```diff
@@ -1610,11 +1610,11 @@ impl fmt::Display for Ipv6Addr {
                 /// Write a colon-separated part of the address
                 #[inline]
                 fn fmt_subslice(f: &mut fmt::Formatter<'_>, chunk: &[u16]) -> fmt::Result {
-                    if let Some(first) = chunk.first() {
-                        fmt::LowerHex::fmt(first, f)?;
-                        for segment in &chunk[1..] {
+                    if let Some((first, tail)) = chunk.split_first() {
+                        write!(f, "{:x}", first)?;
+                        for segment in tail {
                             f.write_char(':')?;
-                            fmt::LowerHex::fmt(segment, f)?;
+                            write!(f, "{:x}", segment)?;
```

Comment:
```rust
                             write!(f, ":{:x}", segment)?;
```
Should we perhaps do this instead?

Reply:
I don't think it makes much difference since the formatting code will call `write_str` on the `":"` separately anyways.

Reply:
I am more concern about the style. However, I don't have a strong opinion on this.

Reply:
I just wonder if there is a performance impact. Both works for me.